observed.py

```diff
@@ -237,13 +241,13 @@ def add_observer(self, observer, identify_observed=False):
 
         # If the observer is a bound method,
         if hasattr(observer, "__self__"):
-            result = self._add_bound_method(observer, identify_observed)
+            result: bool = self._add_bound_method(observer, identify_observed)
         # Otherwise, assume observer is a normal function.
         else:
-            result = self._add_function(observer, identify_observed)
+            result: bool = self._add_function(observer, identify_observed)
```

Comment:
Why are these `bool` annotations needed?

Reply:
hey, it is often helpful for type-checker i.e. mypy, pyre etc. to provide these inline type hinting since otherwise they throw a warning; it also provides more clarity as to what is the variable type if - for any reason - `_add_function` has no `-> ...` typing

Reply:
Ok but isn't it better to add the return type for `_add_function` and *not* add a hint here?